src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3049,13 +3049,18 @@ private void RemoveDirectoryInfoItem(DirectoryInfo directory, bool recurse, bool
             {
                 try
                 {
+                    // TODO:
+                    // Different symlinks seem to vary by behavior.
+                    // In particular, OneDrive symlinks won't remove without recurse,
+                    // but the .NET API here does not allow us to distinguish them.
+                    // We may need to revisit using p/Invokes here to get the right behavior
```

Comment:
Should we remove the comment too?

Reply:
I just put it in in the last commit, since the current behaviour won't work for OneDrive symlinks (why I opened the PR).

Reply:
Maybe it is better to open a tracking issue?

Reply:
How about both? Comments like this often help me when I find them in the code, because I usually hit them when I'm trying to fix something that I didn't realise was related.